src/pal/pal_apple.h

```diff
@@ -35,6 +36,31 @@ namespace snmalloc
      * descriptor for the mapping.
      */
     static constexpr int anonymous_memory_fd = VM_MAKE_TAG(PALAnonID);
+
+    /*
+     * Note : The root's implementation works fine on Intel
+     * however mprotect/PROT_NONE fails on ARM especially
+     * since the 11.2 release (seems known issue spotted
+     * in various projects ; might be a temporary fix).
+     */
+    template<bool page_aligned = false>
+    static void zero(void* p, size_t size) noexcept
+    {
+      ::bzero(p, size);
```

Comment:
If this is required only on AArch64 (is it only AArch64 macOS, or iOS derivatives as well?) then can we guard it on a `constexpr if` check that the AAL is for AArch64, not x86?

Reply:
fair enough. yes Mac M1 is "only" a super iOs device anyway.